lib/ansible/config/base.yml

```diff
@@ -215,537 +224,635 @@ ANSIBLE_SSH_PIPELINING:
+    - When false, the handlers will not run if a failure has occurred on a host.
+    - This can also be set per play or on the command line. See Handlers and Failure for more details.
   env: [{name: ANSIBLE_FORCE_HANDLERS}]
   ini:
   - {key: force_handlers, section: defaults}
   type: boolean
-  yaml: {key: defaults.force_handlers}
+  version_added: "1.9.1"
 DEFAULT_FORKS:
+  name: Number of task forks
   default: 5
   description: Maximum number of forks Ansible will use to execute tasks on target hosts.
   env: [{name: ANSIBLE_FORKS}]
   ini:
   - {key: forks, section: defaults}
   type: integer
-  yaml: {key: defaults.forks}
 DEFAULT_GATHERING:
-    default: 'implicit'
-    description: 'TODO: write it'
-    env: [{name: ANSIBLE_GATHERING}]
-    ini:
-      - key: gathering
-        section: defaults
-    yaml: {key: defaults.gathering}
+  name: Gathering behaviour
+  default: 'implicit'
+  description:
+    - This setting controls the default policy of fact gathering (facts discovered about remote systems).
+    - When ‘implicit’ (the default), the cache plugin will be ignored and facts will be gathered per play unless ‘gather_facts: False’ is set.
+    - When ‘explicit’ the inverse is true, facts will not be gathered unless directly requested in the play.
+    - The ‘smart’ value means each new host that has no facts discovered will be scanned,
+      but if the same host is addressed in multiple plays it will not be contacted again in the playbook run.
+    - This option can be useful for those wishing to save fact gathering time. Both ‘smart’ and ‘explicit’ will use the cache plugin.
+  env: [{name: ANSIBLE_GATHERING}]
+  ini:
+    - key: gathering
+      section: defaults
+  version_added: "1.6"
+  type: list
+  choices: ['all', 'network', 'hardware', 'virtual', 'ohai', 'facter']
 DEFAULT_GATHER_SUBSET:
-    default: 'all'
-    description: 'TODO: write it'
-    env: [{name: ANSIBLE_GATHER_SUBSET}]
-    ini:
-      - key: gather_subset
-        section: defaults
-    yaml: {key: defaults.gather_subset}
+  name: Gather facts subset
+  default: 'all'
+  description:
+      - Set the `gather_subset` option for the M(setup) task in the implicit fact gathering.
+        See the module documentation for specifics.
+      - It does **not** apply to user defined M(setup) tasks.
+  env: [{name: ANSIBLE_GATHER_SUBSET}]
+  ini:
+    - key: gather_subset
+      section: defaults
+  version_added: "2.1"
 DEFAULT_GATHER_TIMEOUT:
+  name: Gather facts timeout
   default: 10
-  description: 'TODO: write it'
+  description:
+    - Set the timeout in seconds for the implicit fact gathering.
+    - It does **not** apply to user defined M(setup) tasks.
   env: [{name: ANSIBLE_GATHER_TIMEOUT}]
   ini:
   - {key: gather_timeout, section: defaults}
   type: integer
   yaml: {key: defaults.gather_timeout}
 DEFAULT_HANDLER_INCLUDES_STATIC:
+  name: Make handler M(include) static
```

Comment:
should the 'M()' macro/marker be allowed in the name?

Reply:
unsure